File: OOP/OOP MOCK/Assignment05.cpp
#include<iostream>
using namespace std;
#define M 5
template <class T>
class Sort{
    T arr[M];
    int n;

    public:

    Sort(){
        getArr();
        cout<< "Before sorting ";
        printArr();
        SelectionSort();
        cout<< "After sorting ";
        printArr();
    }

    void getArr(){
        cout << "Enter size : ";
        cin >>n;

        cout << "Enter elements :\n";
        for(int i=0; i<n; i++){
            cout << "arr[" << i << "] = ";
            cin >> arr[i];
        }
        cout << "Array stored successfully!\n";
    }

    void printArr(){
        cout << " array is : ";

        for (int i=0; i<n; i++)
            cout << arr[i] << " ";

        cout << endl;
    }

    void SelectionSort(){

        cout << "Performing selection sort...\n";
        for(int i=0; i<n-1; i++){

            int minPos = i;

            for(int j=i+1; j<n;j++){
                if (arr[j] < arr[minPos])
                    minPos = j;
            }
            swap(arr[i], arr[minPos]);
        }
        cout << "Selection sort performed successfully!!\n";
    }
};

int main(int argc, char const *argv[])
{
    int ch;

    while (true)
    {
        cout << endl
             << "Enter choice." << endl
             << "Selection for data-type :" << endl
             << "\t1. int" << endl
             << "\t2. char" << endl
             << "\t3. float" << endl
             << "\t4. double" << endl
             << "\t5. string" << endl
             << "PRESS 0 TO EXIT" << endl
             << endl;
            cin >> ch;

        if(ch==0){
            cout << "TERMINATING\n\n";
            exit(0);
        }
        else if(ch==1)
            Sort<int> obj;
        
        else if(ch==2)
            Sort<char> obj;
        
        else if(ch==3)
            Sort<float> obj;
        
        else if(ch==4)
            Sort<double> obj;
        
        else if(ch==5)
            Sort<string> obj;
        
        else 
            cout << "INVALID CHOICE!!\nLE'S TRY AGAIN\n\n";
        
    }

    return 0;
}

File: FDS/Assignment05.py
# selection sort

def selectionSort(arr,n):
    for i in range(n-1):
        minP = i

        for j in range(i+1,n):
            if arr[j] < arr[minP]:
                minP = j
        arr[minP], arr[i] = arr[i], arr[minP]

def bubbleSort(arr,n):
    for i in range(n-1):
        for j in range(n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
arr = [0,6,5,3,2,1,8,9,10]
print(arr)
bubbleSort(arr, len(arr))
print(arr)


''''
OUTPUT:
Enter the number of students: 4
89
90
86
87

 Marks before sorting:  [89, 90, 86, 87]

 1.Selection sort 
 2. Bubble sort
 3.Top five scores

 Enter your choice: 1

 After sorting:
[86, 87, 89, 90]

Enter your choice: 2

 After sorting:
[86, 87, 89, 90]

Enter your choice: 3
[90, 89, 87, 86]
'''



    